File: Whyd/WDPlayer/WDQueuePlayer.m
//
//  WDQueuePlayer.m
//  Whyd
//
//  Created by Damien Romito on 02/08/14.
//  Copyright (c) 2014 Damien Romito. All rights reserved.
//

#import "WDQueuePlayer.h"

@implementation WDQueuePlayer

- (void)insertItem:(AVPlayerItem *)item afterItem:(AVPlayerItem *)afterItem
{
    
    if (![self.items containsObject:item]) {
        
        @try {
            [super insertItem:item afterItem:afterItem];
        }
        @catch (NSException *exception) {
            NSLog(@"EXCEPTION %@",exception);
        }

    }
    else
    {
        DLog(@"ITEM ALREADY CONTAINED");
    }
}

- (void)dealloc
{
    
}

@end

File: Whyd/WDPlayer/WDQueuePlayer.swift
//
//  WDQueuePlayer.swift
//  Whyd
//
//  Created by Damien Romito on 28/07/14.
//  Copyright (c) 2014 Damien Romito. All rights reserved.
//

import UIKit
import AVFoundation

class WDQueuePlayer: AVQueuePlayer {
   //ini
    override func insertItem(item: AVPlayerItem!, afterItem: AVPlayerItem!) {
        
        var exist = false
        
        for i in self.items(){
            if item === i {
                println("swift item \(item)")
                exist = true
            }
            
        }
        if !exist{
            super.insertItem(item, afterItem: afterItem);
        }
    }
}
